src/renderer/components/monaco-editor/monaco-editor.tsx

```diff
@@ -0,0 +1,290 @@
+ */
+
+import styles from "./monaco-editor.module.css";
+import React from "react";
+import { action, computed, makeObservable, observable, reaction } from "mobx";
+import { observer } from "mobx-react";
+import { editor, Uri } from "monaco-editor";
+import { MonacoTheme, registerCustomThemes } from "./monaco-themes";
+import { MonacoValidator, monacoValidators } from "./monaco-validators";
+import { cssNames, disposer, toJS } from "../../utils";
+import { UserStore } from "../../../common/user-store";
+import { ThemeStore } from "../../theme.store";
+import debounce from "lodash/debounce";
+
+registerCustomThemes(); // setup
+
+export type MonacoEditorId = string;
+
+export interface MonacoEditorProps {
+  id?: MonacoEditorId; // associating editor's ID with created model.uri
+  className?: string;
+  style?: React.CSSProperties;
+  autoFocus?: boolean;
+  readOnly?: boolean;
+  theme?: MonacoTheme;
+  language?: "yaml" | "json"; // supported list of languages, configure in `webpack.renderer.ts`
+  options?: Partial<editor.IStandaloneEditorConstructionOptions>; // customize editor's initialization options
+  value?: string;
+  onChange?(value: string, evt: editor.IModelContentChangedEvent): void; // catch latest value updates
+  onError?(error?: Error | unknown): void; // provide syntax validation error, etc.
+  onDidLayoutChange?(info: editor.EditorLayoutInfo): void;
+  onDidContentSizeChange?(evt: editor.IContentSizeChangedEvent): void;
+  onModelChange?(model: editor.ITextModel, prev?: editor.ITextModel): void;
+}
+
+export const defaultEditorProps: Partial<MonacoEditorProps> = {
+  language: "yaml",
+  get theme(): MonacoTheme {
+    // theme for monaco-editor defined in `src/renderer/themes/lens-*.json`
+    return ThemeStore.getInstance().activeTheme.monacoTheme;
+  }
+};
+
+@observer
+export class MonacoEditor extends React.Component<MonacoEditorProps> {
+  static defaultProps = defaultEditorProps as object;
+  static viewStates = new WeakMap<Uri, editor.ICodeEditorViewState>();
+
+  static createUri(id: MonacoEditorId): Uri {
+    return Uri.file(`/monaco-editor/${id}`);
+  }
+
+  public staticId = `editor-id#${Math.round(1e7 * Math.random())}`;
+  public dispose = disposer();
+
+  // TODO: investigate how to replace with "common/logger"
+  //  currently leads for stucking UI forever & infinite loop.
+  //  e.g. happens on tab change/create, maybe some other cases too.
+  logger = console;
+
+  @observable.ref containerElem: HTMLElement;
+  @observable.ref editor: editor.IStandaloneCodeEditor;
+  @observable dimensions: { width?: number, height?: number } = {};
+  @observable unmounting = false;
+
+  constructor(props: MonacoEditorProps) {
+    super(props);
+    makeObservable(this);
+  }
+
+  @computed get model(): editor.ITextModel {
+    const uri = MonacoEditor.createUri(this.props.id ?? this.staticId);
+    const model = editor.getModels().find(model => model.uri.toString() == uri.toString());
```

Comment:
Having investigated the code of monaco-editor. You can safely replace this with `editor.getModel(uri)`